woocommerce/woocommerce-gutenberg-products-block
# Product Gallery: Add Next/Previous Buttons block

This PR adds a Next/Previous Buttons inner block for the Product Gallery block.

For now, there's a temporary placeholder in the editor and the very rudimentary styling in the frontend.

This one is tricky and also a good opportunity to reaccess our strategy of implementation. Should we leverage the current Woo core code for the Gallery or should we build something from scratch.

In the case of the **Next/Previous Buttons**:
- In order for the **Next/Previous** buttons to work the Thumbnails need to be enabled (that allows the markup with all images to load) - which at the moment is in conflict with how the **Large Image** block is set up by default.
- The "easiest" solution of using the built-in filter doesn't produce the desired result:
```
		add_filter(
			'woocommerce_single_product_carousel_options',
			function( $options ) {
				$options['directionNav'] = true;
				return $options;
			}
		);
```

This seems to be affecting _other_ flexsliders on the page, even when used within the buffered context.

- I have worked around this, utilizing the Flexslider API and the buttons work, but the above concerns need to be addressed and should be taken into account.

Note:
There's currently a lint error that's a false positive being triggered by Dashicons, but this is very minor and will be addressed:
`Unexpected missing generic font family  font-family-no-missing-generic-family-keyword`



Closes #9901

### Testing
1. Make sure you have a Product Gallery and Large Image blocks added in the **Single Product** template.
2. Add the Next/Previous Buttons block and verify that the buttons work correctly on the frontend. 

### WooCommerce Visibility
